packages/flutter/lib/src/services/text_editing.dart

```diff
@@ -57,12 +57,25 @@ class TextRange {
   /// The text before this range.
   String textBefore(String text) {
     assert(isNormalized);
+    if(text.isNotEmpty && start > 0 && start < text.length ){
+      final int startCodeUnit = text.codeUnitAt(start);
+      if( startCodeUnit & 0x8C00 == 0x8C00 ){
```

Comment:
Same spacing nit as above.  This also seems dangerous - Why are we checking for these specific bits to be set on the first code unit?  Are we certain that this impacts all unicode characters with those bits set? Could we somehow document what this magic number means?

Reply:
this magic number is utf-16  encoding